services/client/util.go

```diff
@@ -66,3 +69,52 @@ func serviceKey(s services.ServiceID) string {
 	}
 	return fmt.Sprintf(keyFormat, s.Environment(), s.Name())
 }
+
+func placementProtoFromValue(v kv.Value) (*schema.Placement, error) {
+	var placementProto schema.Placement
+	if err := v.Unmarshal(&placementProto); err != nil {
+		return nil, err
+	}
+
+	return &placementProto, nil
+}
+
+func placementFromValue(v kv.Value) (services.Placement, error) {
+	placementProto, err := placementProtoFromValue(v)
+	if err != nil {
+		return nil, err
+	}
+
+	p, err := placement.NewPlacementFromProto(placementProto)
+	if err != nil {
+		return nil, err
+	}
+
+	return p.SetVersion(v.Version()), nil
+}
+
+func placementsProtoFromValue(v kv.Value) (*schema.PlacementSnapshots, error) {
```

Comment:
nit: `stagedPlacementProtoFromValue` might be a better name?

Reply:
gonna rename this one to placementSnapshotsFromValue since that's exactly what this function does.